Rerouting to app/home on button click

leanneeliatra opened the pull request:
### Description
Fix for bug [[BUG] missing roles button doesn't do anything #1490](https://github.com/opensearch-project/security-dashboards-plugin/issues/1490)

### Category
[Bug fix]

### Why these changes are required?
The button on the /customerror/missing-role page was not doing anything, it should bring the user back to the dashboard

### What is the old behavior before changes and new behavior after changes?
**Old behaviour**
1. From the /app/customerror/missing-role screen a button 'Back to OpenSearch dashboards Home ...' did nothing when clicked.

**New behaviour**
1. From the /app/customerror/missing-role screen a button 'Back to OpenSearch dashboards Home ...' routes to the app dashboard when clicked


### Issues Resolved
[[BUG] missing roles button doesn't do anything #1490](https://github.com/opensearch-project/security-dashboards-plugin/issues/1490)

### Testing
in progress.
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

peternied reviewed (changes requested):
Thanks for adding this, before we merge could you create/modify a [test case](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/DEVELOPER_GUIDE.md#testing) that validates the link works?

leanneeliatra commented:
> Thanks for adding this, before we merge could you create/modify a [test case](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/DEVELOPER_GUIDE.md#testing) that validates the link works?

Yes, for sure Peter thanks. They're in progress I just wanted to submit the changes so far in case there were any comments for the change.

cwperks reviewed (commented):
Hi @leanneeliatra , from what I understand of the issue the only real action a user with 0 roles could do would be to logout. Reading the comments on the issue, it sounds like the button on that page should be replaced with a logout button.

https://github.com/opensearch-project/security-dashboards-plugin/issues/1490#issuecomment-1656401599

cwperks commented:
@leanneeliatra Please see my comment above. From the conversation on the issue, this button should be changed to a logout button.

> Hi @leanneeliatra , from what I understand of the issue the only real action a user with 0 roles could do would be to logout. Reading the comments on the issue, it sounds like the button on that page should be replaced with a logout button. Without a logout button the user would have to clear cookies in the browser to logout.

> https://github.com/opensearch-project/security-dashboards-plugin/issues/1490#issuecomment-1656401599

leanneeliatra commented:
> @leanneeliatra Please see my comment above. From the conversation on the issue, this button should be changed to a logout button.
> 
> > Hi @leanneeliatra , from what I understand of the issue the only real action a user with 0 roles could do would be to logout. Reading the comments on the issue, it sounds like the button on that page should be replaced with a logout button. Without a logout button the user would have to clear cookies in the browser to logout.
> 
> > [#1490 (comment)](https://github.com/opensearch-project/security-dashboards-plugin/issues/1490#issuecomment-1656401599)

Thank you Craig, that's complete

leanneeliatra commented:
Hi @peternied I have added the integration tests: https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/830
There are unit tests covering the logout functionality so that's covered from that aspect.
& I changed the functionality to a logout button cc @cwperks 
Thank you all.

derek-ho commented on public/apps/customerror/custom-error.tsx:
```diff
@@ -45,7 +45,7 @@ export function CustomErrorPage(props: CustomErrorDeps) {
       <EuiSpacer size="s" />
-      <EuiButton fill href={props.http.basePath.serverBasePath} fullWidth>
+      <EuiButton fill onClick={() => logout(props.http, '')} fullWidth>
         Back to OpenSearch Dashboards Home
```
Do we want to change this text to "Logout"?

cwperks replied:
+1, otherwise this change looks good to me

leanneeliatra replied:
Complete

derek-ho reviewed (approved):
Other than button text change looks good. Can we change the base branch to main and backport to 2.x?

derek-ho commented:
This should also fix: https://github.com/opensearch-project/security-dashboards-plugin/issues/842

leanneeliatra reviewed (commented):
I have added the integration tests @peternied 
https://github.com/opensearch-project/opensearch-dashboards-functional-test/pull/830

peternied reviewed (dismissed):
Thanks for adding the tests.

peternied commented:
I just saw that this PR was pointed to the 2.8 branch, it should go into main, please rebase against main

leanneeliatra commented:
> I just saw that this PR was pointed to the 2.8 branch, it should go into main, please rebase against main

I will indeed thanks @peternied

leanneeliatra commented:
> I just saw that this PR was pointed to the 2.8 branch, it should go into main, please rebase against main

Rebase to main is complete, thanks.

peternied commented:
I think this is a good time to recommend looking at https://ohshitgit.com/ - the harsh language aside, this is a fairly quick and well written guide to solving git history problems, like the one you look to be in the middle of, give that a try and I'd recommend reaching out to someone else online via the public slack for troubleshooting if I'm not around.

peternied commented on .github/workflows/prerequisite-test.yml:
```diff
@@ -0,0 +1,14 @@
+name: Prerequisite Checks
```
Revert this change

leanneeliatra replied:
complete by rewriting the changes to this branch (with the changes from main & the edit to the button needed to close this ticket).

peternied commented on DEVELOPER_GUIDE.md:
```diff
@@ -54,7 +54,6 @@ nvm use --install
 
```
Revert this change

leanneeliatra replied:
complete by rewriting the changes to this branch (with the changes from main & the edit to the button needed to close this ticket).

peternied commented on public/apps/configuration/panels/get-started.tsx:
```diff
@@ -188,7 +188,6 @@ export function GetStarted(props: AppDependencies) {
 
```
Revert this change

leanneeliatra replied:
complete by rewriting the changes to this branch (with the changes from main & the edit to the button needed to close this ticket).

peternied commented on release-notes/opensearch-security-dashboards-plugin.release-notes-2.8.0.0.md:
```diff
@@ -10,4 +10,4 @@ Compatible with Opensearch-Dashboards 2.8.0
 * Add new cluster permissions constants for long-running operation notification feature in Index-Management repo ([#1444](https://github.com/opensearch-project/security-dashboards-plugin/pull/1444))
-* Adds the newly created admin api permissions to the static dropdown list ([#1426](https://github.com/opensearch-project/security-dashboards-plugin/pull/1426))
\ No newline at end of file
+* Adds the newly created admin api permissions to the static dropdown list ([#1426](https://github.com/opensearch-project/security-dashboards-plugin/pull/1426))
```
Revert this change

peternied reviewed (commented):
Looks like there are still a few unrelated changes in this pull request, please revert those files

leanneeliatra commented:
Added the changes to a new [branch](https://github.com/opensearch-project/security-dashboards-plugin/compare/main...leanneeliatra:security-dashboards-plugin-fork:Switch-the-home-link-to-logout-on-security-error-pages), branched from `main` as directed in the call yesterday.

leanneeliatra commented:
PR was failing on unit tests, during development I could see the logout button funtcitonality was alreaedy covered so didnt' think unit tests would be needed but to cover the line as specified 
`public/apps/customerror/custom-error.tsx#L49
Added line #L49 was not covered by tests`
I am adding the unit tests now (in progress). Thanks.

scrawfor99 commented:
Hey @leanneeliatra, to easily fix the issues you are facing, I would recommend these steps:

1. Create a new branch off of main using `git checkout main`, then `git branch switchHomeLink`
2. Checkout that new branch 
3. Manually copy the changes you made here or use `git cherry-pick <commit id>` to grab those changes 
4. Overwrite this base with a force push using `git push -u origin 2.8 --force`  

This should remove the compatibility issues you are facing.

peternied commented:
@leanneeliatra Looks like there are still some test failures going on, how is that going?

leanneeliatra commented:
Can someone assist with the errors on the CI pipeline please?
The various cypress tests are failing to connect to chrome
```
Running:  plugins/security-dashboards-plugin/inaccessible_tenancy_features.js             (1 of 1)
Still waiting to connect to Chrome, retrying in 1 second (attempt 18/62)
```

leanneeliatra commented:
> @leanneeliatra Looks like there are still some test failures going on, how is that going?

Hi Peter, great. Glad to say they are done now and the CI is passing for the tests added and coverage is increaced.

Now I just need to get this over the line. The Cypress tests are failing on Chrome, was it the security team you said I could reach out to there or have I misunderstood. I will get on to the team who can fix this ASAP. Thanks.

peternied commented:
@leanneeliatra I've re-triggered CI test execution, I saw socket failures which would be completely unrelated to your change.  @RyanL1997 could you confirm the state of the cypress tests?

Don't worry about that code coverage issue - I am not concerned about that line getting coverage.

leanneeliatra commented:
> @leanneeliatra I've re-triggered CI test execution, I saw socket failures which would be completely unrelated to your change. @RyanL1997 could you confirm the state of the cypress tests?
> 
> Don't worry about that code coverage issue - I am not concerned about that line getting coverage.

Thanks a million Peter.

leanneeliatra commented:
> Hi @leanneeliatra and @peternied . I think the socket failure is related to this issue: [cypress-io/cypress-documentation#5483](https://github.com/cypress-io/cypress-documentation/issues/5483). So it is not from our end. But I can try to fix it by rolling back the chrome driver version.

Thanks for looking into this @RyanL1997 and yes, please do that. It would be great to get this merged in! Thanks.

leanneeliatra commented:
> Update: frontend pipeline should be fixed, so I'm re-running all checks.

leanneeliatra commented:
I am pulling in Ryans updates to the CI that will fix my checks on this PR and which will also simultaneously rerun the checks.

leanneeliatra commented:
The main branch has been merged in to my own branch but the CI is still failing, I will reach out to @RyanL1997 to discuss.

leanneeliatra commented:
Myself and @RyanL1997 are in communication about this.

leanneeliatra commented:
@RyanL1997 is going to have a look at the cypress tests for me (thank you :) )

RyanL1997 commented:
Reference to https://github.com/opensearch-project/security-dashboards-plugin/issues/1599, the manual testing results:
<img width="732" alt="Screenshot 2023-10-11 at 9 24 53 PM" src="https://github.com/opensearch-project/security-dashboards-plugin/assets/109499885/82f17f3d-e78e-4520-b1bc-5d29fb3a3774">
<img width="722" alt="Screenshot 2023-10-11 at 9 28 03 PM" src="https://github.com/opensearch-project/security-dashboards-plugin/assets/109499885/804c396e-97b0-4579-8020-2ac9c8a431e5">
<img width="733" alt="Screenshot 2023-10-11 at 9 28 56 PM" src="https://github.com/opensearch-project/security-dashboards-plugin/assets/109499885/e3d8ea90-2d24-426a-9531-b8239646180a">

leanneeliatra commented:
Thanks a million @RyanL1997 I really appreciate that!

Can I please get final reviews when you have a moment @cliu123 @scrawfor99 and @davidlago please.

leanneeliatra commented:
Can this be merged please?

peternied commented:
@RyanL1997 could you look into the CI issue?

leanneeliatra commented:
Can this be merged please! :)

leanneeliatra commented:
> @leanneeliatra I've re-triggered CI test execution, I saw socket failures which would be completely unrelated to your change. @RyanL1997 could you confirm the state of the cypress tests?
> 
> Don't worry about that code coverage issue - I am not concerned about that line getting coverage.

Please merge ASAP. Thank you.
